drivers/input/input_chsc5x.c

```diff
@@ -170,6 +166,76 @@ static int chsc5x_init(const struct device *dev)
 		k_msleep(1);
 	}
 
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+int chsc5x_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct chsc5x_config *config = dev->config;
+	int ret;
+
+	if (config->reset_gpio.port == NULL) {
+		return -ENOTSUP;
+	}
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		ret = chsc5x_reset(dev);
+		break;
+
+	case PM_DEVICE_ACTION_SUSPEND: {
+		const uint8_t write_buffer[] = {
+			CHSC5X_BASE_ADDR1,
+			CHSC5X_BASE_ADDR2,
+			CHSC5X_BASE_ADDR3,
+			0x00,
+			0xF7,
+			0x16,
+			0x05,
+			0x00,
+			0x01,
+			0x00,
+			0x00,
+			0x00,
+			0x00,
+			0x00,
+			0x00,
+			0x00,
+			0x00,
+			0x00,
+			0x03,
+			0xE9,
```

Comment:
that's a lot of magic here,  any chance of cleaning this up somehow? add some defines or at minimum write it a more compact way, guess you are using clang-format, just add  a `clang-format off` `clang-format on` block from the range if that's the case